File: src/main/java/fr/adrienbrault/idea/symfony2plugin/assetMapper/dict/AssetMapperModule.java
package fr.adrienbrault.idea.symfony2plugin.assetMapper.dict;

import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
public record AssetMapperModule(
    @NotNull MappingFileEnum sourceType,
    @NotNull VirtualFile sourceFile,
    @NotNull String key,
    @Nullable String path,
    @Nullable String url,
    @Nullable String version,
    @Nullable Boolean entrypoint,
    @Nullable String type
) {
}

File: src/main/kotlin/fr/adrienbrault/idea/symfony2plugin/assetMapper/dict/AssetMapperModule.kt
package fr.adrienbrault.idea.symfony2plugin.assetMapper.dict

import com.intellij.openapi.vfs.VirtualFile

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
@JvmRecord
data class AssetMapperModule(
    val sourceType: MappingFileEnum,
    val sourceFile: VirtualFile,
    val key: String,
    val path: String?,
    val url: String?,
    val version: String?,
    val entrypoint: Boolean?,
    val type: String?,
)
